### PR TITLE
Work on authn documentation.

### PR DESCRIPTION
**Pull request by PremyslCerny**

### CODE REVIEW

**Comment by PremyslCerny** [comment]
@ateska Please see and comment.

**Comment by PremyslCerny** [comment]
@SukiCZ Hi, what do you think about this documentation? Is it understandable for you from the user perspective?

File: doc/asab/web/authn.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+`Authorization: Bearer <OAUTH-SERVER-ID>-<ACCESS_TOKEN>`
+
+The following example illustrates how to register a middleware inside the web server container with OAuth 2.0
+implementation and GitHub method.
+
+.. code:: python
+
+    container.WebApp.middlewares.append(
+        asab.web.authn.authn_middleware_factory(self,
+            "oauth2client",
```
Comment (SukiCZ) [inline doc/asab/web/authn.rst]: @PremyslCerny Could this be with comment like
```
            "oauth2client",  # "pubkeyauth" or "basicauth"
```

Comment (PremyslCerny) [reply doc/asab/web/authn.rst]: @SukiCZ Sure, good point. :)

**Comment by PremyslCerny** [comment]
@ateska Please merge. 👍

**Pull request merged**